src/program/mod.rs

```diff
@@ -367,4 +387,29 @@ DECLARE ec BIT
         // program after being re-parsed and serialized.
         assert!(program1.lines().eq(program2.lines()));
     }
+
+    #[test]
+    fn test_get_qubits() {
+        let input = "
+DECLARE ro BIT
+MEASURE q ro
+JUMP-UNLESS @end-reset ro
+X q
+LABEL @end-reset
+
+DEFCAL I 0:
+    DELAY 0 1.0
+DEFFRAME 0 \"rx\":
+    HARDWARE-OBJECT: \"hardware\"
+DEFWAVEFORM custom:
+    1,2
+I 0
+";
+        let program = Program::from_str(input).unwrap();
+        let expected = vec![Qubit::Fixed(0), Qubit::Variable("q".to_string())]
+            .into_iter()
+            .collect::<HashSet<_>>();
+        let actual = program.get_used_qubits();
+        assert_eq!(expected, actual);
+    }
```

Comment:
I may be missing it, but consider also include a qubit operation in the test Quil program that would _not_ be returned from the `get_used_qubits` method to check that this behavior is correct and continues to be later.

Reply:
Thinking back on this, I'm not sure my request makes sense! Happy to connect on this offline 😆

Reply:
Yeah, not sure I can think of a case where that would happen.